zerver/lib/export.py

```diff
@@ -1425,7 +1425,10 @@ def do_import_realm(import_dir):
 
     convert_to_id_fields(data, 'zerver_realm', 'notifications_stream')
     fix_datetime_fields(data, 'zerver_realm')
-    realm = Realm(**data['zerver_realm'][0])
+    zr = data['zerver_realm'][0]
+    # Convert the authentication methods from list of (name, bool) to bitfield
+    zr["authentication_methods"] = int(''.join(['1' if a[1] else '0' for a in zr["authentication_methods"]]), 2)
```

Comment:
We don't really do this sort of inline Python hackery.  You should write a function for this fixup job, and ideally add some unit tests for it.

Reply:
Updated but haven't added test. I think doing the export/import circle should suffice as a test.